RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -146,15 +207,54 @@ namespace {
 
     tofpid.gammasq_pi = 1. + magp2 * m_pi_inv2;
     tofpid.beta_pi = std::sqrt(1. - 1. / tofpid.gammasq_pi);
-    tofpid.dt_pi = tofpid.pathlength / tofpid.beta_pi * c_inv;
+    switch (choice) {
+      case TofCalc::cost:
+        tofpid.dt_pi = tofpid.pathlength / tofpid.beta_pi * c_inv;
+        break;
+      case TofCalc::segm:
+        tofpid.dt_pi = trs.computeTof(m_pi_inv2);
+        break;
+      case TofCalc::mixd:
+        tofpid.dt_pi = trs.computeTof(m_pi_inv2) + tofpid.pathlength / tofpid.beta_pi * c_inv;
+        break;
+      default:
+        tofpid.dt_pi = 1;
+        break;
```

Comment:
```suggestion
```
seems unnecessary considering the definition of `TofCalc`

Reply:
in practice it is, it is just a protection against possible not foreseen values. Given the local use of the class I believe it is mostly an academic concern. I'll take your suggestion.